libosrmc/osrmc.cc

```diff
@@ -208,46 +250,86 @@ void osrmc_table_params_add_source(osrmc_table_params_t params, size_t index, os
   auto* params_typed = reinterpret_cast<osrm::TableParameters*>(params);
   params_typed->sources.emplace_back(index);
 } catch (const std::exception& e) {
-  *error = new osrmc_error{e.what()};
+  osrmc_error_from_exception(e, error);
 }
 
 void osrmc_table_params_add_destination(osrmc_table_params_t params, size_t index, osrmc_error_t* error) try {
   auto* params_typed = reinterpret_cast<osrm::TableParameters*>(params);
   params_typed->destinations.emplace_back(index);
 } catch (const std::exception& e) {
-  *error = new osrmc_error{e.what()};
+  osrmc_error_from_exception(e, error);
+}
+
+void osrmc_table_params_set_annotations(osrmc_table_params_t params, osrmc_table_annotations_t annotations, osrmc_error_t* error) try {
+  auto* params_typed = reinterpret_cast<osrm::TableParameters*>(params);
+  auto* annotations_typed = reinterpret_cast<osrm::TableParameters::AnnotationsType*>(annotations);
+  params_typed->annotations = *annotations_typed;
+} catch (const std::exception& e) {
+  osrmc_error_from_exception(e, error);
 }
 
+struct osrmc_table_response {
+  osrm::json::Object json;
+  osrm::TableParameters::AnnotationsType annotations;
```

Comment:
Why do we store the annotations in the response object here? So that we can pass the annotations along since we don"t have a global `Table` object?

Reply:
I really wanted to avoid this, but at the time it was the only way I could see to produce an error if someone tried to get a distance from a table response that hadn't requested the distance. 

I think I was having a bad day yesterday, as I see now I don't need to check for this explicitly: we can [check whether the key is set in the response object](https://github.com/Project-OSRM/osrm-backend/blob/master/include/engine/api/table_api.hpp#L78-L88). I'll switch to doing that, and get rid of this extra `osrmc_table_response` struct.

Reply:
Ah, I didn't think of this, too! Good catch, let's do it this way then!